### DISCUSSION TITLE
Cannot read properties of undefined (reading 'length')

### DISCUSSION THREAD

**Discussion by sebastianrath** [webpack/webpack#16218] [Q&A]
Hey there,

I am upgrading my webpack dependency in my Electron app from `4.46.0` to the latest `5.74.0`. I am running into the following error message when building my project:

```
⠇ Generating browser application bundles (phase: building).../
       /../node_modules/webpack/lib/util/hash/wasm-hash.js:108
                const length = data.length;

TypeError: Cannot read properties of undefined (reading 'length')
    at WasmHash._updateWithBuffer (/.../node_modules/webpack/lib/util/hash/wasm-hash.js:108:23)
    at WasmHash.update (/.../node_modules/webpack/lib/util/hash/wasm-hash.js:51:8)
    at BatchedHash.update (/.../node_modules/webpack/lib/util/hash/BatchedHash.js:50:14)
    at /.../node_modules/webpack/lib/FileSystemInfo.js:3464:11
    at processQueue (/.../node_modules/webpack/lib/util/processAsyncTree.js:55:4)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
```

https://github.com/webpack/webpack/blob/9fcaa243573005d6fdece9a3f8d89a0e8b399613/lib/util/hash/wasm-hash.js#L106-L108

This is the content of the `webpack.config.js`:

```JavaScript
const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin');
const NodePolyfillPlugin = require("node-polyfill-webpack-plugin");

const path = require('path');

let sassImplementation = require('sass');

module.exports = {
	output: {
		path: path.resolve(__dirname, 'build-js'),
		filename: 'app.js'
	},
  externals: {
    'fsevents': "require('fsevents')"
  },
	plugins: [new MonacoWebpackPlugin(),
    new NodePolyfillPlugin({
      excludeAliases: ["console"]
    })
  ],
  target: 'electron-renderer',
  module: {
    rules: [
      {
        test: /\.css$/i,
				use: ['style-loader', 'css-loader']
      },
      {
        test: /\.scss$/,
        use: [
          'postcss-loader',
          {
            loader: 'sass-loader',
            options: {
              implementation: sassImplementation,
              sourceMap: false,
              sassOptions: {
                precision: 8,
              },
            },
          },
        ],
      },
    ],
  },
};
```

Does anything stand out that could cause this, or is this an issue in webpack? Any help is highly appreciated.

Thanks!

**Comment by alexander-akait** [comment]
I think a bug somewhere in `monaco-editor-webpack-plugin`. Does `monaco-editor-webpack-plugin` use the latest `webpack-sources`?

**Reply by sebastianrath** [reply]
After some digging, it could be related to https://github.com/nodejs/node-gyp/issues/2713 (also see this PR https://github.com/nodejs/node-gyp/pull/2721).

TLDR: New(er) versions of `node-gyp` create a symlink of Python at `./node_modules/{dependency}/build/build/node_gyp_bins/python3`. As far as I can tell, webpack tries to read the content and returns with `undefined` as the symlink isn't followed.

**Reply by sebastianrath** [reply]
I tried several options to exclude these files but no chance. Any suggestion which `exclude` field in my webpack.config.js should work best?

**Reply by alexander-akait** [reply]
I'm afraid it's impossible to say without more information. Why you think the problem is in `node-gyp`? What is Node.js version?

**Reply by alexander-akait** [reply]
Try to comment `node-polyfill-webpack-plugin` too and check it

**Reply by sebastianrath** [reply]
> I'm afraid it's impossible to say without more information. Why you think the problem is in `node-gyp`? What is Node.js version?

After deleting all mentioned `Python` symbolic links inside `node_modules` the bundling succeeds successfully. Node is `v16.15.0`.

> Try to comment `node-polyfill-webpack-plugin` too and check it

Same error.

**Reply by alexander-akait** [reply]
What package do you build when you are getting this error?

**Reply by sebastianrath** [reply]
I am building an Electron app that has a couple of native module dependencies like [balena-io-modules/drivelist](https://github.com/balena-io-modules/drivelist), and a few private ones. All of them are built with the latest `node-gyp: v9.1.0` and contain the symbolic links after building.

**Reply by alexander-akait** [reply]
Really weird, maybe you can create small repo with example of the problem so we can investigate it

**Reply by yoyo837** [reply]
What a coincidence, I was able to reproduce this issue. 😅

**Reply by yoyo837** [reply]
<img width="993" alt="image" src="https://user-images.githubusercontent.com/6134547/230031985-d86da898-8eee-4344-aa99-76d494564066.png">

**Reply by alexander-akait** [reply]
It means you have old webpack plugin which doesn't update own code to v5

**Reply by adonis-lsh** [reply]
me too

**Reply by lyztroy** [reply]
i have the same problem. what should I do ?

**Reply by alexander-akait** [reply]
Run `npm ls webpack`

**Comment by artemis-prime** [comment]
I'm seeing this in next 14.1.3... seemingly out of nowhere

**Reply by artemis-prime** [reply]
And in Next 14.2.3! what the heck??

**Reply by faustinozanetto** [reply]
I experienced the same issue in 14.2.3 and deleting the .next folder and building again fixed it for me.

**Reply by avallonazevedo** [reply]
@artemis-prime @faustinozanetto -- have you guys found a solution? I have been facing this issue since yesterday in version 14.2.3. Tried multiple potential solutions, but the problem persists.

**Reply by faustinozanetto** [reply]
My solution as I said was deleting the .next folder and executing next dev again like normally. It was fixed for my but maybe your code has another bug that causes it.

**Reply by ahmedrowaihi** [reply]
> My solution as I said was deleting the .next folder and executing next dev again like normally. It was fixed for my but maybe your code has another bug that causes it.

It's insane, delete node_modules us needed every 3 hot reloads..

**Reply by jeanhdev** [reply]
Found the problem in my case - make sure all statically generated pages are generated successfully. Example - for me it was the CMS API called when building the paths and props that wasn't running and triggered the error. Found this by actually digging in.

**Reply by ahmedrowaihi** [reply]
I noticed this problem only happens with webpack

using tubropack ( --turbo )
Doesn't show the issue

**Reply by phinpho** [reply]
I'm seeing this on Next.JS 14.2.16

One thing that I managed to do to mitigate the issue was changing the minified file javascript `node_modules/next/dist/compiled/webpack/bundle5.js`:

Using a regular expression search on this file, I tried `[^.]_updateWithBuffer(\s*)\(([^0-9]\w)*\)(\s*)\{`.

Then, I changed the string found from `}_updateWithBuffer(v){` to `}_updateWithBuffer(v){if(v===undefined)return;`

Before:
```
...)$.copyWithin(0,v,L)}}_updateWithBuffer(v){const{exports:E,buffered:P,mem:R}=this;const $=v...
```

After:
```
...)$.copyWithin(0,v,L)}}_updateWithBuffer(v){if(v===undefined)return;const{exports:E,buffered:P,mem:R}=this;const $=v...
```

**Reply by alexander-akait** [reply]
@phinpho can you provide stack trace?

**Reply by phinpho** [reply]
@alexander-akait, I'm redacting some details for security reasons, but I think it will be fine for analyzing the Stack Trace.

When I run the Next.js dev server (very early on), the latest line of the command output is ` ✓ Ready in 1526ms`. When I try to access the page with the path `/[slug]/page` in the browser, it shows the following logs and the page gets stuck. The latest message in the logs is ` ○ Compiling /404 ...`, anything else (except if I interrupt the process).

The command:
```
npm run dev
```

In the `package.json`, the property `scripts.dev` has the following value:
```
"NODE_OPTIONS='--trace-deprecation' next dev"
```

The result:
```
  ▲ Next.js 14.2.16
  - Local:        http://localhost:3000
  - Experiments (use with caution):
    · externalDir
    · scrollRestoration
    · forceSwcTransforms

 ✓ Starting...
 ✓ Ready in 3s
 ✓ Compiled /src/middleware in 162ms (142 modules)
 ○ Compiling /[slug]/page ...
 ✓ Compiled /[slug]/page in 4s (994 modules)
(node:57213) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
    at node:punycode:3:9
    at BuiltinModule.compileForInternalLoader (node:internal/bootstrap/realm:399:7)
    at BuiltinModule.compileForPublicLoader (node:internal/bootstrap/realm:338:10)
    at loadBuiltinModule (node:internal/modules/helpers:96:7)
    at Module._load (node:internal/modules/cjs/loader:1070:17)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Module.require (node:internal/modules/cjs/loader:1304:12)
    at mod.require ([[--PROJECT-PATH--]]/node_modules/next/dist/server/require-hook.js:65:28)
    at require (node:internal/modules/helpers:123:16)
    at Object.<anonymous> ([[--PROJECT-PATH--]]/node_modules/node-fetch/node_modules/whatwg-url/lib/url-state-machine.js:2:18)
    at Module._compile (node:internal/modules/cjs/loader:1504:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1588:10)
    at Module.load (node:internal/modules/cjs/loader:1282:32)
    at Module._load (node:internal/modules/cjs/loader:1098:12)
TypeError: Cannot read properties of undefined (reading 'length')
    at WasmHash._updateWithBuffer ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1384455)
    at WasmHash.update ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1383790)
    at BatchedHash.update ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1380151)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:297115
    at processQueue ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1397772)
    at processAsyncTree ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1397778)
    at FileSystemInfo._resolveContextTimestamp ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:296723)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:270356
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1333530
    at Hook.eval [as callAsync] (eval at create ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:13:28858), <anonymous>:6:1)
    at AsyncQueue._handleResult ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1333084)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1332935
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:296661
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:295397
    at done ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/neo-async/async.js:1:10300)
 ⨯ uncaughtException: TypeError: Cannot read properties of undefined (reading 'length')
    at WasmHash._updateWithBuffer ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1384455)
    at WasmHash.update ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1383790)
    at BatchedHash.update ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1380151)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:297115
    at processQueue ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1397772)
    at processAsyncTree ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1397778)
    at FileSystemInfo._resolveContextTimestamp ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:296723)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:270356
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1333530
    at Hook.eval [as callAsync] (eval at create ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:13:28858), <anonymous>:6:1)
    at AsyncQueue._handleResult ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1333084)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1332935
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:296661
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:295397
    at done ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/neo-async/async.js:1:10300)
 ⨯ uncaughtException: TypeError: Cannot read properties of undefined (reading 'length')
    at WasmHash._updateWithBuffer ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1384455)
    at WasmHash.update ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1383790)
    at BatchedHash.update ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1380151)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:297115
    at processQueue ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1397772)
    at processAsyncTree ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1397778)
    at FileSystemInfo._resolveContextTimestamp ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:296723)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:270356
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1333530
    at Hook.eval [as callAsync] (eval at create ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:13:28858), <anonymous>:6:1)
    at AsyncQueue._handleResult ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1333084)
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:1332935
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:296661
    at [[--PROJECT-PATH--]]/node_modules/next/dist/compiled/webpack/bundle5.js:28:295397
    at done ([[--PROJECT-PATH--]]/node_modules/next/dist/compiled/neo-async/async.js:1:10300)
 ○ Compiling /404 ...
```
Node Version:
```
22.5.1
```

NPM Version:
```
10.8.2
```

Versions of some packages found in the `package-lock.json` file:
| Package                                                      | Version   |
|--------------------------------------------------------------|-----------|
| `node_modules/neo-async`                                     | 2.6.2     |
| `node_modules/next`                                          | 14.2.16   |
| `node_modules/webpack`                                       | 5.95.0    |
| `node_modules/webpack-bundle-analyzer`                       | 4.10.1    |
| `node_modules/webpack-dev-middleware`                        | 6.1.3     |
| `node_modules/webpack-hot-middleware`                        | 2.26.1    |
| `node_modules/webpack-sources`                               | 3.2.3     | 
| `node_modules/webpack-virtual-modules`                       | 0.5.0     |
| `node_modules/unplugin`                                      | 1.14.1    |
| `node_modules/unplugin/node_modules/webpack-virtual-modules` | 0.6.2     |

_Let me know if you need more information_

**Reply by alexander-akait** [reply]
`webpack-virtual-modules` can be a problem, when trying to create a hash, but for some reason our content is `undefined`, here the place - https://github.com/webpack/webpack/blob/main/lib/util/hash/wasm-hash.js#L42, I asked for the stack so I could see where it came from, but apparently it happens in the hook so the stack is dirty, https://github.com/webpack/webpack/blob/main/lib/FileSystemInfo.js#L3582 - the main problem here is how to we work with directories, you can debug which directory is a problem, hashes are - https://github.com/webpack/webpack/blob/main/lib/FileSystemInfo.js#L3607 and https://github.com/webpack/webpack/blob/main/lib/FileSystemInfo.js#L3613 

Can you debug? Maybe we should add `undefined` checks, but this may lead to the fact that the directory change will not be tracked by the timestamp. Need a little more information

**Reply by alexander-akait** [reply]
To debug we need `entry` value and `hashes` - https://github.com/webpack/webpack/blob/main/lib/FileSystemInfo.js#L3606

**Reply by yue4u** [reply]
I just encountered a similar error using storybook

```
  entry: {
    safeTime: 1732269935475,
    timestampHash: '6fa9cbf877b15bff2026d0cfa34364cf',
    hash: '2c6dac90603fc802b664d11e672aad55',
    symlinks: Set(1) {
      '/Users/<redacted>'
    }
  },
  hashes: [ '1c7c4e56b8455609a99c91f0940bfef8', undefined, undefined ]
}
/Users/<redacted>/node_modules/@storybook/cli/bin/index.js:23
  throw error;
  ^

TypeError: Cannot read properties of undefined (reading 'length')
    at WasmHash._updateWithBuffer (/Users/<redacted>/node_modules/webpack/lib/util/hash/wasm-hash.js:108:23)
    at WasmHash.update (/Users/<redacted>/node_modules/webpack/lib/util/hash/wasm-hash.js:51:8)
    at BatchedHash.update (/Users/<redacted>/node_modules/webpack/lib/util/hash/BatchedHash.js:50:14)
    at /Users/<redacted>/node_modules/webpack/lib/FileSystemInfo.js:3518:12
    at processQueue (/Users/<redacted>/node_modules/webpack/lib/util/processAsyncTree.js:61:4)
    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)

Node.js v22.11.0
```

**Reply by alexander-akait** [reply]
@yue4u Can you create a reproducible repository?

**Comment by aheritier** [comment]
We faced the issue on our CI in august and it just reappears.
We were thinking it could be due to some invalid symbolic links in node_modules but today we don't find any of them. it's hard to create a testcase to reproduce the issue because it happens only on our CI and it could be related with the cache management. 
Not sure how to debug this issue.

**Reply by alexander-akait** [reply]
@aheritier Can you show me stacktrace?

**Reply by aheritier** [reply]
Sure @alexander-akait 

```
/opt/actions-runner/_work/acme/acme/node_modules/webpack/lib/util/hash/wasm-hash.js:108
		const length = data.length;
		                    ^

TypeError: Cannot read properties of undefined (reading 'length')
    at WasmHash._updateWithBuffer (/opt/actions-runner/_work/acme/acme/node_modules/webpack/lib/util/hash/wasm-hash.js:108:23)
    at WasmHash.update (/opt/actions-runner/_work/acme/acme/node_modules/webpack/lib/util/hash/wasm-hash.js:51:8)
    at BatchedHash.update (/opt/actions-runner/_work/acme/acme/node_modules/webpack/lib/util/hash/BatchedHash.js:53:14)
    at /opt/actions-runner/_work/acme/acme/node_modules/webpack/lib/FileSystemInfo.js:3518:11
    at processQueue (/opt/actions-runner/_work/acme/acme/node_modules/webpack/lib/util/processAsyncTree.js:61:4)
    at processTicksAndRejections (node:internal/process/task_queues:77:11)
    at runNextTicks (node:internal/process/task_queues:64:3)
    at process.processImmediate (node:internal/timers:449:9)

Node.js v20.14.0
```

**Reply by alexander-akait** [reply]
The same... It looks like some unofficial plugin is implementing something but not updating the hash. Are you using nextjs?

**Reply by alexander-akait** [reply]
Can you debug this place - https://github.com/webpack/webpack/discussions/16218#discussioncomment-5529636

**Reply by aheritier** [reply]
A theory we had in the past is that the we had an invalid symlink pointing to nothing. In August it's by removing such anomaly that we thought we solved the issue. Sadly I verified on our current node_modules folder and there is no broken link. 

We are not using nextjs but react, now with the number of dependencies we have on our monolith it is highly possible that we use something deprecated plugins. I am looking at your link

**Reply by alexander-akait** [reply]
@aheritier As another solution, you can try to create a reproducible repository or give me private access to the stag project (just send invite) (I can sign an NDA) and I will see why this is happening

**Reply by aheritier** [reply]
`npm ls webpack` gives me this:

```
acme@ /Users/arnaud/Workspace/dev/acme/acme
├─┬ @pmmmwh/react-refresh-webpack-plugin@0.5.15
│ └── webpack@5.91.0 deduped
├─┬ @storybook/react-webpack5@8.1.1
│ ├─┬ @storybook/builder-webpack5@8.1.1
│ │ ├─┬ fork-ts-checker-webpack-plugin@8.0.0
│ │ │ └── webpack@5.91.0 deduped
│ │ ├─┬ html-webpack-plugin@5.6.0
│ │ │ └── webpack@5.91.0 deduped
│ │ ├─┬ style-loader@3.3.4
│ │ │ └── webpack@5.91.0 deduped
│ │ ├─┬ webpack-dev-middleware@6.1.3
│ │ │ └── webpack@5.91.0 deduped
│ │ └── webpack@5.91.0 deduped
│ └─┬ @storybook/preset-react-webpack@8.1.1
│   ├─┬ @storybook/react-docgen-typescript-plugin@1.0.6--canary.9.0c3f3b7.0
│   │ └── webpack@5.91.0 deduped
│   └── webpack@5.91.0 deduped
├─┬ @webpack-cli/serve@2.0.5
│ └── webpack@5.91.0 deduped
├─┬ babel-loader@9.2.1
│ └── webpack@5.91.0 deduped
├─┬ circular-dependency-plugin@5.2.2
│ └── webpack@5.91.0 deduped
├─┬ css-loader@6.10.0
│ └── webpack@5.91.0 deduped
├─┬ css-minimizer-webpack-plugin@6.0.0
│ └── webpack@5.91.0 deduped
├─┬ imports-loader@5.0.0
│ └── webpack@5.91.0 deduped
├─┬ mini-css-extract-plugin@2.9.2
│ └── webpack@5.91.0 deduped
├─┬ postcss-loader@8.1.1
│ └── webpack@5.91.0 deduped
├─┬ sass-loader@16.0.3
│ └── webpack@5.91.0 deduped
├─┬ style-loader@4.0.0
│ └── webpack@5.91.0 deduped
├─┬ terser-webpack-plugin@5.3.10
│ └── webpack@5.91.0 deduped
├─┬ webpack-assets-manifest@5.2.1
│ └── webpack@5.91.0 deduped
├─┬ webpack-cli@5.1.4
│ ├─┬ @webpack-cli/configtest@2.1.1
│ │ └── webpack@5.91.0 deduped
│ ├─┬ @webpack-cli/info@2.0.2
│ │ └── webpack@5.91.0 deduped
│ └── webpack@5.91.0 deduped
├─┬ webpack-dev-server@5.1.0
│ ├─┬ webpack-dev-middleware@7.4.2
│ │ └── webpack@5.91.0 deduped
│ └── webpack@5.91.0 deduped
├── webpack@5.91.0
└─┬ zipper@89.8.0 -> ./packages/zipper
  ├─┬ raw-loader@4.0.2
  │ └── webpack@5.91.0 deduped
  ├─┬ ts-loader@9.5.1
  │ └── webpack@5.91.0 deduped
  └── webpack@5.91.0 deduped
```

We will continue to investigate with our teammates tomorrow

**Reply by alexander-akait** [reply]
I don't see anything very dangerous...

Maybe you can try to create a reproducible repository? Just comment lines step by step and look when it happend

**Comment by gitmahin** [comment]
I faced this same issue in Next.js 15, but solved it by deleting node_modules and reinstalling everything.

**Reply by yiheee-lab** [reply]
True but it keep happening and make things super frustrated.

**Comment by yiheee-lab** [comment]
Well for Nextjs with tauri, try disable WASM.

next.config.ts
```typescript
const nextConfig: NextConfig = {
  webpack: (config) => {
    config.experiments = {
      ...config.experiments,
      asyncWebAssembly: false,
      syncWebAssembly: false,
    };
    return config;
  },
};
```
